models/user.go

```diff
@@ -1208,6 +1208,16 @@ func deleteUser(e Engine, u *User) error {
 	// ***** END: PublicKey *****
 
 	// ***** START: GPGPublicKey *****
+	keys, err := ListGPGKeys(u.ID, ListOptions{})
```

Comment:
need `listGPGKeys(e Engine, ...)` version

Reply:
To clarify, is that the `listGPGKeys` defined [here?](https://github.com/go-gitea/gitea/blob/master/routers/api/v1/user/gpg_key.go#L18-L31)

Reply:
I'll send you the diff to apply :)

Reply:
```diff
diff --git a/models/gpg_key.go b/models/gpg_key.go
index b944fdcbf..6e5347bac 100644
--- a/models/gpg_key.go
+++ b/models/gpg_key.go
@@ -65,7 +65,11 @@ func (key *GPGKey) AfterLoad(session *xorm.Session) {
 
 // ListGPGKeys returns a list of public keys belongs to given user.
 func ListGPGKeys(uid int64, listOptions ListOptions) ([]*GPGKey, error) {
-       sess := x.Where("owner_id=? AND primary_key_id=''", uid)
+       return listGPGKeys(x, uid, listOptions)
+}
+
+func listGPGKeys(e Engine, uid int64, listOptions ListOptions) ([]*GPGKey, error) {
+       sess := e.Where("owner_id=? AND primary_key_id=''", uid)
        if listOptions.Page != 0 {
                sess = listOptions.setSessionPagination(sess)
        }
diff --git a/models/user.go b/models/user.go
index 642d9d56a..495fed1ff 100644
--- a/models/user.go
+++ b/models/user.go
@@ -1208,7 +1208,7 @@ func deleteUser(e Engine, u *User) error {
        // ***** END: PublicKey *****
 
        // ***** START: GPGPublicKey *****
-       keys, err := ListGPGKeys(u.ID, ListOptions{})
+       keys, err := listGPGKeys(e, u.ID, ListOptions{})
        if err != nil {
                return fmt.Errorf("ListGPGKeys: %v", err)
        }
```

Reply:
```diff
diff --git a/models/gpg_key.go b/models/gpg_key.go
index b944fdcbf..3e8ddd962 100644
--- a/models/gpg_key.go
+++ b/models/gpg_key.go
@@ -65,7 +65,11 @@ func (key *GPGKey) AfterLoad(session *xorm.Session) {
 
 // ListGPGKeys returns a list of public keys belongs to given user.
 func ListGPGKeys(uid int64, listOptions ListOptions) ([]*GPGKey, error) {
-       sess := x.Where("owner_id=? AND primary_key_id=''", uid)
+       return listGPGKeys(x, uid, listOptions)
+}
+
+func listGPGKeys(e Engine, uid int64, listOptions ListOptions) ([]*GPGKey, error) {
+       sess := e.Table(&GPGKey{}).Where("owner_id=? AND primary_key_id=''", uid)
        if listOptions.Page != 0 {
                sess = listOptions.setSessionPagination(sess)
        }
diff --git a/models/user.go b/models/user.go
index 642d9d56a..495fed1ff 100644
--- a/models/user.go
+++ b/models/user.go
@@ -1208,7 +1208,7 @@ func deleteUser(e Engine, u *User) error {
        // ***** END: PublicKey *****
 
        // ***** START: GPGPublicKey *****
-       keys, err := ListGPGKeys(u.ID, ListOptions{})
+       keys, err := listGPGKeys(e, u.ID, ListOptions{})
        if err != nil {
                return fmt.Errorf("ListGPGKeys: %v", err)
        }
```